pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -82,42 +82,58 @@ func (pc *PodCache) event(obj interface{}, ev model.Event) error {
 		case model.EventAdd:
 			switch pod.Status.Phase {
 			case v1.PodPending, v1.PodRunning:
-				// add to cache if the pod is running or pending
-				pc.keys[ip] = key
+				if _, ok := pc.podsByIP[ip]; !ok {
+					// add to cache if the pod is running or pending
+					pc.podsByIP[ip] = key
+					pc.configUpdates(ip)
+				}
 			}
 		case model.EventUpdate:
 			if pod.DeletionTimestamp != nil {
 				// delete only if this pod was in the cache
-				if pc.keys[ip] == key {
-					delete(pc.keys, ip)
+				if pc.podsByIP[ip] == key {
+					delete(pc.podsByIP, ip)
 				}
 				return nil
 			}
 			switch pod.Status.Phase {
 			case v1.PodPending, v1.PodRunning:
-				// add to cache if the pod is running or pending
-				pc.keys[ip] = key
+				if _, ok := pc.podsByIP[ip]; !ok {
+					// add to cache if the pod is running or pending
+					pc.podsByIP[ip] = key
+					pc.configUpdates(ip)
+				}
+
 			default:
 				// delete if the pod switched to other states and is in the cache
-				if pc.keys[ip] == key {
-					delete(pc.keys, ip)
+				if pc.podsByIP[ip] == key {
+					delete(pc.podsByIP, ip)
 				}
 			}
 		case model.EventDelete:
 			// delete only if this pod was in the cache
-			if pc.keys[ip] == key {
-				delete(pc.keys, ip)
+			if pc.podsByIP[ip] == key {
+				delete(pc.podsByIP, ip)
 			}
 		}
 	}
 	return nil
 }
 
+func (pc *PodCache) configUpdates(ip string) {
+	if pc.c != nil && pc.c.XDSUpdater != nil {
+		pc.c.XDSUpdater.ConfigUpdate(&model.PushRequest{
+			Full:          true,
+			TargetProxies: map[string]struct{}{pc.c.ClusterID + ip: {}},
```

Comment:
why not just directly enqueue the single proxy to the push queue? I think we do that in other places

Reply:
Nope, `TargetProxies` is a new field i added, the reason i add clusterID here is to distinguish multi cluster proxies.

Reply:
is it feasible to change this? it makes it a lot more complicated to have 2 ways to trigger a push to a single proxy

Reply:
I searched and find no send to pushQueue directly now, I think removed with WorkloadUpdate.

Reply:
Hm.. can we just extend the XDSUpdater api with ProxyUpdate that does the pushQueue.Enquueue? I don't think it makes sense to add all this plumbing to push to a single proxy?

Reply:
ok, this can reduce the node labels locality latency. And i will try to think how we can get rid of direct dependency on k8s.